server/service/src/report/report_service.rs

```diff
@@ -325,16 +324,88 @@ fn get_report(ctx: &ServiceContext, id: &str) -> Result<Report, RepositoryError>
 
 fn query_reports(
     ctx: &ServiceContext,
-    pagination: Option<PaginationOption>,
     filter: Option<ReportFilter>,
     sort: Option<ReportSort>,
 ) -> Result<Vec<Report>, ListError> {
+    let app_version: Version = Version::from_package_json();
+
     let repo = ReportRepository::new(&ctx.connection);
-    let pagination = get_default_pagination(pagination, MAX_LIMIT, MIN_LIMIT)?;
     let filter = filter
         .unwrap_or_default()
         .r#type(ReportType::OmSupply.equal_to());
-    Ok(repo.query(pagination, Some(filter.clone()), sort)?)
+
+    let reports_to_show_meta_data = report_filter_method(
+        repo.query_meta_data(Some(filter.clone()), None)?,
+        app_version,
+    );
+
+    let filter = ReportFilter::new().id(EqualFilter::equal_any(
+        reports_to_show_meta_data
+            .into_iter()
+            .map(|r| r.id)
+            .collect(),
+    ));
+
+    let reports = repo.query(Some(filter), sort)?;
+    Ok(reports)
+}
+
+fn report_filter_method(
```

Comment:
I have a feeling this logic could be more concise. However, after taking some time to think about it, I figure it is not worth spending time on now given we are only passing around report_meta_data

Reply:
This is great, really easy to read and performant, thank you. There is a lot to like about this, abstraction for readability, the `reports_of_code and `custom_reports_of_code` reduces the cognitive load (one step at a time)

The only improvement I can think of, the query_meta_data could return a domain object, with a little bit of mapping after `.load`, to get the version number (from_str) and store it into that domain object (reduce mapping in compare_major_minor and in find_latest_report)